processor/metricsgenerationprocessor/config.go

```diff
@@ -0,0 +1,178 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricsgenerationprocessor
+
+import (
+	"fmt"
+	"sort"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+const (
+	// NewMetricFieldName is the mapstructure field name for NewMetricName field
+	NewMetricFieldName = "name"
```

Comment:
Please rename the variable names to better match the new names.

Actually, we also want to make sure not to add too much to public API. If these names seem helpful you can make them private (lowercase first letter), otherwise consider inlining them

Reply:
There are a lot of public APIs in this PR, please go through and make all that can be private private

Reply:
Pushed an update.